crates/revm/src/executor.rs

```diff
@@ -474,6 +487,11 @@ pub fn commit_state_changes<DB>(
                 // the account already exists and its storage was cleared, preserve its previous
                 // state
                 AccountState::StorageCleared
+            } else if has_state_clear_eip &&
+                matches!(cached_account.account_state, AccountState::NotExisting) &&
```

Comment:
Does this do anything, we are checking if `cached_account.account_state == AccountState::NotExisting` and then setting it to `NotExisting`?

Reply:
more precisely, important that we do not destroy just any empty account, only the ones that were indeed created before state clear EIP

Reply:
But logic does nothing, what it does is:
```
cached_acccount.account_state = if cached_acccount.account_state == AccountState::NotExisting {
     AccountState::NotExisting
}
```

Reply:
Aha, you are short `circuit` the else branch

Reply:
the desired behavior is covered in `test_touched_unchanged_account` test. you can comment out this logic and run that test

Reply:
It makes sense, I was just looking at `else if` logic and miss `else`